ArcGISRuntimeSDKQt_CppSamples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.cpp

```diff
@@ -213,3 +221,34 @@ void AddDynamicEntityLayer::purgeAllObservations()
   auto future = m_dynamicEntityDataSource->purgeAllAsync();
   Q_UNUSED(future)
 }
+
+void AddDynamicEntityLayer::identifyLayerAtMouseClick(const QMouseEvent &e)
+{
+  // Hide the callout (if it is already hidden this will do nothing)
+  m_mapView->calloutData()->setVisible(false);
+  // Reseting m_tempParent gives it a new parent and cleans up any previously owned children like IdentifyLayerResult and DynamicEntity objects
+  m_tempParent.reset(new QObject(this));
```

Comment:
Pick one ownership model; QScopedPointer or 'this'.

Reply:
used this.